### PR TITLE
 New map design with three areas in one map.

### PR DESCRIPTION
**Pull request by Howard-12**
# Description

Refactor the  code to allow generate three area in the map
Added barriers to lock the next area (there is a open temporary entry to the next area, walk straight up to get to the entry).
The current map dimension is 160x144 in terms of tiles. Each area is 160x48,  the player starting point is at 80x10.
```
| ---------------------|                O (player)
|                      | 
|                      |     AIR(160x48)
| --------   ----------|
|                      |
|                      |     WATER(160x48)
| --------   ----------|
|                      |
|            O         |     FOREST(160x48)
| ---------------------|
```

Fixes / Closes # (issue)
None

## Type of change
- [x] Map design

# How Has This Been Tested?

Run around as the player make sure there is not empty spot in the map also tiles should connected correctly as expected.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by BaileyS03 — APPROVED** [review]
image
I noticed items spawning on top of each other which I found to be new? In Terrain component it would be good to remove lines 293 - 304, 348-353 or remove from commit to main. Other than this code looks fine and the game play seemed good :)

**Review by mitsudz — APPROVED** [review]
Looks pretty good, however, are you able to update in a future pr for sprint 3 that the newChunks when calculated should not include chunks outside of the border? Also the random position radii in ForestGameArea should be updated to use +- ChunkSize/2.

Aside from this please don't leave tests commented out - either remove them or update them to match the current code.

File: source/core/src/main/com/csse3200/game/areas/ForestGameArea.java [hunk 1]
```diff
@@ -104,13 +116,106 @@ public void create() {
...
 
       //Initialise inventory and quests with loaded data
       player.getComponent(InventoryComponent.class).loadInventoryFromSave();
       player.getComponent(QuestManager.class).loadQuests();
   }
 
+  /**
+   * Unlock an area of the map
+   */
+  private void unlockArea(String area) {
```
Comment (LucasRyall) [inline source/core/src/main/com/csse3200/game/areas/ForestGameArea.java:130]: What is the point of this function (is this a part that hasn't been implemented yet)?

**Review by LucasRyall — APPROVED** [review]
Code looks pretty good apart from that one random funciton, some issues like the world border being a void and items spawning in said void, but these can be fixed in sprint 4.

**Pull request merged**